ipam/allocator.go

```diff
@@ -51,9 +52,11 @@ type Allocator struct {
 	nicknames        map[router.PeerName]string   // so we can map nicknames for rmpeer
 	pendingAllocates []operation                  // held until we get some free space
 	pendingClaims    []operation                  // held until we know who owns the space
+	dead             map[string]time.Time         // containers we heard were dead, and when
```

Comment:
I think my preference would be to turn owned into a map[string]struct, and have that struct contain the list of ips and the time the container was stopped.  This would have less chance of the dead and the owned maps becoming out of sync with future changes.